### PR TITLE
SDK-94: workflow

### PR DESCRIPTION
**Pull request by vol4onok**

### CODE REVIEW

File: src/Presentation/Console/Commands/InitSdkCommand.php [hunk 1]
```diff
@@ -71,6 +72,31 @@ public function __construct(
...
 
+    /**
+     * @return void
+     */
+    protected function configure()
+    {
+        parent::configure();
+        $this->createDatabase();
+
+        $settings = $this->yamlParser->parseFile($this->settingsPath)['settings'];
```
Comment (miroshnikov-spryker) [inline src/Presentation/Console/Commands/InitSdkCommand.php:83]: There is similar logic in InitProjectCommand. How do you think, is there a point in extracting it to abstract class or trait?

**Review by miroshnikov-spryker — APPROVED** [review]

File: src/Core/Appplication/Service/ProjectWorkflow.php [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        }
+
+        $taskId = $context->getTask()->getId();
+
+        $this->currentWorkflow = $this->workflows->get($this->currentProjectWorkflow, $this->currentProjectWorkflow->getWorkflow());
+        $enabledTransactions = $this->currentWorkflow->getEnabledTransitions($this->currentProjectWorkflow);
+        $enabledTasksIds = [];
+        $metaWorkflow = $this->currentWorkflow->getMetadataStore();
+        foreach ($enabledTransactions as $enabledTransaction) {
+            $transactionTaskId = $metaWorkflow->getTransitionMetadata($enabledTransaction)['task'] ?? null;
```
Comment (Kyrylo-Meretskyi) [inline src/Core/Appplication/Service/ProjectWorkflow.php:93]: The same as here https://github.com/spryker-sdk/sdk/pull/71/files#r804514854

File: src/Core/Appplication/Service/ProjectWorkflow.php [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    /**
+     * @var \Symfony\Component\Workflow\Workflow|null
+     */
+    protected ?Workflow $currentWorkflow = null;
+
+    /**
+     * @var \Symfony\Component\Workflow\Transition|null
+     */
+    protected ?Transition $currentTransaction = null;
```
Comment (Kyrylo-Meretskyi) [inline src/Core/Appplication/Service/ProjectWorkflow.php:51]:
```suggestion
    protected ?Transition $currentTransition = null;
```

File: src/Extension/Settings/Initializers/ProjectUuidInitializer.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        } else {
+            $setting->setValues($this->uuid());
+            $this->projectSettingRepository->save($setting);
+        }
+    }
+
+    /**
+     * @return string
+     */
+    protected function uuid(): string
```
Comment (Kyrylo-Meretskyi) [inline src/Extension/Settings/Initializers/ProjectUuidInitializer.php:49]: I think this method should be in the separated class

File: src/Infrastructure/Entity/Workflow.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    /**
+     * @param string $project
+     *
+     * @return void
+     */
+    public function setProject(string $project): void
+    {
+        $this->project = $project;
```
Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Entity/Workflow.php:46]: Please add return $this for consistency of setters

File: src/Infrastructure/Repository/WorkflowRepository.php [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    public function findOne(string $project): ?WorkflowInterface
+    {
+        $criteria = [
+            'project' => $project,
+        ];
+
+        /** @var \SprykerSdk\Sdk\Infrastructure\Entity\Workflow|null $workflow */
+        $workflow = $this->findOneBy($criteria);
+
+        if ($workflow === null) {
```
Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Repository/WorkflowRepository.php:64]:
```suggestion
        return $workflow;
```

File: src/Extension/Settings/Initializers/ProjectUuidInitializer.php (hunk 4 above)
Comment (vol4onok) [reply src/Extension/Settings/Initializers/ProjectUuidInitializer.php:49]: why? it's related to this Initializer

File: src/Infrastructure/Repository/WorkflowRepository.php (hunk 6 above)
Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Repository/WorkflowRepository.php:64]:
```suggestion
       return $workflow;
```

Comment (Kyrylo-Meretskyi) [inline src/Infrastructure/Repository/WorkflowRepository.php:64]: I think the condition is redundant, $workflow can be null or object

File: src/Core/Appplication/Service/ProjectWorkflow.php [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        return $context;
+    }
+
+    /**
+     * @return array<string>
+     */
+    public function getAll(): array
+    {
+        return array_map(
+            fn (Workflow $workflow) => $workflow->getName(),
```
Comment (Kyrylo-Meretskyi) [inline src/Core/Appplication/Service/ProjectWorkflow.php:155]:
```suggestion
            fn (Workflow $workflow): string => $workflow->getName(),
```

**Review by Kyrylo-Meretskyi — APPROVED** [review]

**Pull request merged**